classes/task/failed_task.php

```diff
@@ -31,6 +31,98 @@ public function get_name() {
         return get_string('taskfailed', 'logstore_xapi');
     }
 
+    private function extract_failed_events($limitnum) {
+        global $DB;
+        $conditions = null;
```

Comment:
Code styling: Could you add extra empty line after globals, please?

Reply:
I have fixed this and will commit today.